src/web_interface.py

```diff
@@ -0,0 +1,253 @@
+    st.title("Average Team Score")
+
+    team_score = data_processor.calculate_team_score(dictionary, 0.75, 0.25, 0.5)
+
+    st.write("The calculated average team score for this repo is: ", team_score)
+
+    return team_score
+
+
+def graph_issues(dictionary):
+    """Graphs the issues modified of individuals for web interface."""
+    st.title("Issues Contributed To By An Individual")  # disp`aly relevant
+
+    df = pd.DataFrame.from_dict(dictionary, orient="index").T
+
+    columns = st.multiselect(
+        label="Enter the names of specific contributors below:", options=df.columns
+    )  # allow users to display specific contributor information on dataframe graph
+
+    for name in columns:
+        issues_commented = len(df[name][8])
+        df[name][8] = issues_commented
+        issues_opened = len(df[name][9])
+        df[name][9] = issues_opened
+
+    st.bar_chart(
+        df[columns][8:10]
+    )  # display dataframe/graph that vizualizes commit info
+
+    return df
+
+
+def graph_pull_request(dictionary):
+    """Graph PRs contributed to by an individual for web interface."""
+    st.title("Pull Requests Contributed to By An Individual")
+
+    df = pd.DataFrame.from_dict(dictionary, orient="index").T
+
+    columns = st.multiselect(
+        label="Enter the names of specific contributors below:", options=df.columns
+    )  # allow users to display specific contributor information on dataframe graph
+    for name in columns:
+        prs_commented = len(df[name][10])
+        df[name][10] = prs_commented
+        prs_opened = len(df[name][11])
+        df[name][11] = prs_opened
+
+    st.bar_chart(
+        df[columns][10:12]
+    )  # display dataframe/graph that vizualizes commit info
+
+    return df
+
+
+def graph_test_contributions(dictionary):
+    """Graph test contributions for web interface."""
+    st.title("Team Members Who Contribute Source Code Without Tests")
+
+    df = pd.DataFrame.from_dict(dictionary, orient="index").T
+
+    columns = st.multiselect(
+        label="Enter the names of specific contributors below:", options=df.columns
+    )  # allow users to display specific contributor information on dataframe graph
+    st.bar_chart(
+        df[columns][12:14]
+    )  # display dataframe/graph that vizualizes commit info
+
+    return df
+
+
+def graph_percent_individual_contribution(dictionary):
+    """Graph percentage of individual contribution."""
+    st.title("Team Members Who Contribute Source Code Without Tests")
+    print("Old dict")
+    data_collection.print_individual_in_table(
+        data_dict=dictionary,
+        headings=["COMMITS", "ADDED", "REMOVED", "MODIFIED", "RATIO", "FORMAT"],
+    )
+    print("\n ##################### \n")
+    print("new dict")
+    new_dict = data_processor.individual_contribution(dictionary)
+    data_collection.print_individual_in_table(
+        data_dict=new_dict,
+        headings=["COMMITS", "ADDED", "REMOVED", "MODIFIED", "RATIO", "FORMAT"],
+    )
```

Comment:
We still need to rework this so it displays a table of the percentages dictionary.